src/nn.h

```diff
@@ -352,6 +364,7 @@ NN_EXPORT int nn_setsockopt (int s, int level, int option, const void *optval,
     size_t optvallen);
 NN_EXPORT int nn_getsockopt (int s, int level, int option, void *optval,
     size_t *optvallen);
+NN_EXPORT int nn_setopt (int option, const void *optval, size_t optvallen);
```

Comment:
Again, not sure if a run-time function is better than compile-time configuration option ... do there exist use cases where it even makes sense to set at run-time?